### PR TITLE
Remove != and == from SequencePosition

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/dotnet/corefx/issues/29001

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
Sorry. I just don't buy that this is a good solution to the problem we have. I think we have a problem with ROS, not with SequencePosition.

**Comment by pakrym** [comment]
Considering that any code that uses these operators almost certainly has a bug in it and absence of conversation or way to fix in in ROS removing them is a safe bet. Operators could be added later any time.

**Comment by KrzysztofCwalina** [comment]
I think we have a real problem that we should fix. Removing these operators just sweeps the problem under the rug.

**Comment by pakrym** [comment]
Ok, what' is your proposed fix?

**Comment by KrzysztofCwalina** [comment]
I think @ahsonkhan should meet with you offline and come up with a recommendation. 

cc: @joshfree

**Comment by davidfowl** [comment]
> I think we have a real problem that we should fix. Removing these operators just sweeps the problem under the rug.

Agreed. The *real* problem is that we expose methods that let you incorrectly use the SequencePosition type. We now have several instances of the same recurring problem where people are incorrectly using sequence position for parsing and it kinda works depending on how you use it (until it doesn't). We didn't decide to remove this randomly, we had buggy code that didn't pass code reviews because we (@pakrym and I) understood that using these methods on SequencePosition is dangerous.

**Comment by pakrym** [comment]
Talked to @KrzysztofCwalina  in person, he said he is fine with the change. I'll file ASK mode request.

**Comment by KrzysztofCwalina** [comment]
Sorry, I added no marge back as I noticed @stephentoub added it, not I.

**Comment by GrabYourPitchforks** [comment]
Just a quick heads up, `new SequencePosition(42, 42).Equals(new SequencePosition(42, 42))` evaluates to `false`. Is this intentional?

**Comment by pakrym** [comment]
Not intentional but also not important for any of current scenarios.

**Comment by GrabYourPitchforks** [comment]
@pakrym Might it be important for future scenarios? The reason I ask is that I don't think the behavior can be changed once the type ships, so I wanted to make sure this was given consideration. "By design" is also a valid answer as long as some thought was given to it. :)

**Comment by pakrym** [comment]
I think calling `.Equals` on the object is more future-proof and doesn't hurt us now.

File: src/System.Memory/src/System/SequencePosition.cs [hunk 1]
```diff
@@ -41,7 +41,9 @@ public SequencePosition(object @object, int integer)
         /// Indicates whether the current <see cref="SequencePosition"/> is equal to another <see cref="SequencePosition"/>.
         /// <see cref="SequencePosition"/> equality does not guarantee that they point to the same location in <see cref="System.Buffers.ReadOnlySequence{T}" />
         /// </summary>
-        public bool Equals(SequencePosition other) => _integer == other._integer && _object == other._object;
+        public bool Equals(SequencePosition other) =>
+            _integer == other._integer &&
+            (_object == null ? other._object == null : _object.Equals(other._object));
```
Comment (GrabYourPitchforks) [inline src/System.Memory/src/System/SequencePosition.cs]: This should just be `... && Object.Equals(this._object, other._object)`.

Comment (GrabYourPitchforks) [inline src/System.Memory/src/System/SequencePosition.cs]: This should just be `... && Object.Equals(this._object, other._object)`.

**Comment by joshfree** [comment]
@pakrym please merge this when ready and also port this to the release/2.1 branch for RC.

**Pull request merged**